### DISCUSSION TITLE
Is it possible to add input to reports?

### DISCUSSION THREAD

**Discussion by k3x** [netbox-community/netbox#9588] [Ideas]
Hi,
we would like to generate a with an input field, so we can limit the report to a specific tennant for example. Is that possible?
Or list all IPs of a subnet as another example.

Thanks
Felix

**Comment by jeremypgn** [comment]
Hi,

I would appreciate some help on this as well.

**Comment by candlerb** [comment]
Reports don't accept input - although it would be nice if they could.

For reports which require the user to select something, I use "Custom Scripts" instead. There is an example [here](https://github.com/netbox-community/customizations/blob/master/scripts/power_summary.py).

It's helpful to include

```
    class Meta:
        scheduling_enabled = False
        commit_default = False
```

to reduce the clutter on the input page.
